reconcile/test/external_resources/test_secrets_sync.py

```diff
@@ -93,3 +96,37 @@ def test_current_new_data_dont_triggers_update(
 ) -> None:
     current.body["data"]["new_key"] = "new_value"
     assert SecretHelper.compare(current, desired) is True
+
+
+def test_vault_secrets_reconciler_read_secret_preserves_kv_v2_cached_data() -> None:
+    """Test that _read_secret doesn't mutate cached data"""
+    original_secret_data = {
+        "db_host": "example.com",
+        "db_password": "secret123",
+        SECRET_UPDATED_AT: "2025-11-20T17:24:32Z",
+    }
+    # Mock secret reader to return the same object (simulating KV v2 cache behavior)
+    mock_secrets_reader = Mock()
```

Comment:
use auto spec instead of raw mock

```python
mock_secrets_reader = create_autospec(SecretReaderBase)
```

Reply:
updated in [c333e87](https://github.com/app-sre/qontract-reconcile/pull/5306/commits/c333e872b5fce1dd4e91011b1e25d863bf7185a9)